### DISCUSSION TITLE
Get recordset data into new ComboBox??

### DISCUSSION THREAD

**Discussion by KJo587** [TomSchimansky/CustomTkinter#198] [General]
I just can't figure out how to get the rows returned from a query into the drop down list of the new ComboBox. Can anyone provide an example?? I can execute the fetchall(), but how do I get it into the list of values?? Code below shows what I have tried, see comments. 

```python
    def get_descriptions(pattern):
        if pattern == "":
            query = "SELECT Description FROM Parts ORDER BY Description"
        else:
            pattern = pattern + "%'"
            query = "SELECT Description FROM Parts WHERE Description LIKE '"+ pattern +" ORDER BY Description"
        
        print(query)       # just to see it
        rows = cursor.execute(query).fetchall()
        if len(rows)==0:
            print("No Records Returned")
            bad_beep()
        else:
            #tlist = list()       # this didn't work
            #for i in rows:
            #    tlist.append(i)
            desc_cb.configure(value=rows)      # this throws an exception: unknown option "-value"

    #create ComboBox
    desc_cb = customtkinter.CTkComboBox(window, width=300,command=get_descriptions,variable=desc_cb_val)
    desc_cb.place(x=400,y=10)
    get_descriptions("")


```

**Comment by felipetesc** [comment]
I would:
1)
```python
# change rows for db_rows
db_rows = cursor.execute(query).fetchall()
```
2)
```python
# change i for db_row
for db_row in db_rows:
```
3)
```python
# change i for db_row
tlist.append(db_row)
```
4)
```python
 desc_cb.configure(rows=db_rows.len) 
```
And see what happens

**Reply by KJo587** [reply]
Running STEP4, I get : 
   unknown option "-rows" 

Thanks for trying!

**Reply by felipetesc** [reply]
sure ! my bad ! I misplaced configure with grid

**Comment by KJo587** [comment]
OK, I stumbled accross a solution, here are the relevant lines:

`        
        rows = cursor.execute(query).fetchall()

        if len(rows)==0:
            print("No Records Returned")
            bad_beep()
        else:
            print("Fill Description ComboBox")
            my_values = [row[0] for row in rows]
            desc_cb.configure(values=my_values)
`

Works like a charm!
